Packs/IronscalesEventCollector/Integrations/IronscalesEventCollector/IronscalesEventCollector.py

```diff
@@ -73,6 +87,115 @@ def get_open_incident_ids(self) -> List[int]:
             or []
         )
 
+    ################## Convert Options for Debugging #####################
+
+    def convert_time_percent_format(self, time):
+        """
+        The API receive timestamps only in iso format, percent encoded
+        """
+        demisto.debug("Test-IronScales: Going in convert_time_percent_format")
+        time = time.isoformat()  # convert to iso format
+        demisto.debug("Test-IronScales: time format to ISO success")
+        time_encoded = quote(time, safe="")  # Percent-encode (e.g., encode '+' to '%2B')
+        demisto.debug("Test-IronScales: quote func success")
+        return time_encoded
+
+    def convert_time_iso_format(self, time):
+        demisto.debug("Test-IronScales: Going in convert_time_iso_format")
+        time = time.isoformat()  # convert to iso format
+        demisto.debug("Test-IronScales: time format to ISO success")
+        return time
+
+    def convert_time_iso_format_with_z(self, time):
+        demisto.debug("Test-IronScales: Going in convert_time_iso_format_with_z")
+        time = time.isoformat().replace("+00:00", "Z")  # convert to iso format
+        demisto.debug("Test-IronScales: time format to ISO with z success")
+        return time
+
+    def do_not_change(self, time):
+        demisto.debug("Test-IronScales: Going in do_not_change")
+        return time.strftime("%Y-%m-%dT%H:%M:%SZ")
+
+    ############ End Debugging Area #################
+
+    def get_all_incident_ids(self, start_time: datetime, max_fetch: int, last_id=None) -> List[int]:
+        """
+        Summary:
+            Pull all the incident IDs from the API, using pagination mechanizm with respect to max_fetch
+        Args:
+            start_time: from what time to start pulling
+            max_fetch: max number of incidents to fetch
+            last_id: last id from the last run
+        Returns:
+            List of incident IDs
+        """
+        demisto.debug("Test-IronScales: going in get_all_incident_ids")
+        conversion_dict = {
+            "convert_time_percent_format": self.convert_time_percent_format,
+            "convert_time_iso_format": self.convert_time_iso_format,
+            "convert_time_iso_format_with_z": self.convert_time_iso_format_with_z,
+            "do_not_change": self.do_not_change,
+        }
+        for func_name, convert_func in conversion_dict.items():
+            try:
+                demisto.debug(f"Test-IronScales: trying to convert with {func_name}")
+                demisto.debug("Test-IronScales: going in get_all_incident_ids")
+                curr_time = datetime.now(timezone.utc)  # Get the current datetime with UTC timezone
+                demisto.debug(f"Test-IronScales: curr_time is {str(curr_time)}")
+                curr_time = convert_func(curr_time)
+                demisto.debug("Test-IronScales: curr time converted successfully. converting start time...")
+                start = convert_func(start_time)
```

Comment:
this wont work if you already have the str time from the last fetches (in case this is not the first fetch)